tests/snippets/strings.py

```diff
@@ -431,3 +431,21 @@ def try_mutate_str():
 assert f'{1234567890.1234:_.2f}' == '1_234_567_890.12'
 with AssertRaises(ValueError, msg="Unknown format code 'd' for object of type 'float'"):
     f'{5.0:04d}'
+
+# Test % formatting
+assert f'{10.0:%}' == '1000.000000%'
+assert f'{10.0:.2%}' == '1000.00%'
+assert f'{10.0:.8%}' == '1000.00000000%'
+assert f'{-10.0:%}' == '-1000.000000%'
+assert f'{-10.0:.2%}' == '-1000.00%'
+assert f'{-10.0:.8%}' == '-1000.00000000%'
+assert '{:%}'.format(float('nan')) == 'nan%'
+assert '{:%}'.format(float('NaN')) == 'nan%'
+assert '{:%}'.format(float('NAN')) == 'nan%'
```

Comment:
The difference in casing is kinda redundant, as the formatting machinery isn't really involved; it's `float.__new__` that parses the `nan`s.

Reply:
Indeed it is, will remove :+1: